src/model/comment.ml

```diff
@@ -130,15 +130,31 @@ and link_content_of_inline_elements l =
   l |> List.map link_content_of_inline_element |> List.concat
 
 let find_zero_heading docs : link_content option =
-  let rec find_map f = function
-    | [] -> None
-    | x :: l -> (
-        match f x with Some _ as result -> result | None -> find_map f l)
-  in
-  find_map
+  Odoc_utils.List.find_map
     (fun doc ->
       match doc.Location_.value with
       | `Heading ({ heading_level = `Title; _ }, _, h_content) ->
           Some (link_content_of_inline_elements h_content)
       | _ -> None)
     docs
+
+let extract_frontmatter docs : _ =
+  let parse_frontmatter s =
+    let lines = Astring.String.cuts ~sep:"\n" s in
+    Odoc_utils.List.filter_map
+      (fun line -> Astring.String.cut ~sep:":" line)
+      lines
+  in
+  let fm, content =
+    let fm, rev_content =
+      List.fold_left
```

Comment:
One advantage of the frontmatter is that it's immediately recognizable. With this syntax, it can be confused for a code block and present several times all over the document.
This doesn't sound much better than `@` tags, which would be shorter to type and more easily recognized.

I think a proper frontmatter should be a new syntax and expected at the beginning of the page only. For example using the markdown syntax (but without the yaml):
```
---
foo: bar
---
{0 Title}
```

Reply:
We've agreed to consider this, but not to hold up the PR until it's decided, as several things depend upon this. We'll revisit the question before releasing 3.0.